tests/pos/dependent-annot.scala

```diff
@@ -0,0 +1,7 @@
+class C
+class ann(x: Any*) extends annotation.Annotation
+
+def f(y: C, z: C) =
+  def g(): C @ann(y, z) = ???
```

Comment:
the printing has changed here, now instead of 
```scala
def g(): C @ann([y,z : Any]*)
```
we get
```scala
def g(): C new ann([y,z : Any]*) = ???
```